src/Verilator.cpp

```diff
@@ -563,15 +574,32 @@ static void verilate(const string& argString) {
 
     V3Error::abortIfWarnings();
 
+    if (v3Global.hierPlanp()) {  // This run is for just write a makefile
+        UASSERT(v3Global.opt.hierarchical(), "must be set");
```

Comment:
```suggestion
        UASSERT(v3Global.opt.hierarchical(), "hierarchical must be set");
```

(In the unlikely case one of these fires it's helpful when the user posts the error to have a better idea what broke as the file line numbers might be hard to determine e.g. from an older version or something they edited.)